main.cpp

```diff
@@ -334,39 +334,58 @@ int StatCompare(const CAddrReport& a, const CAddrReport& b) {
   }
 }
 
+bool isDumpDbRunning = false;
```

Comment:
I don't believe this function is signal safe: https://man7.org/linux/man-pages/man7/signal-safety.7.html

In particular, it indirectly performs memory allocations. Generally the approach is setting just a flag in the handler itself, and having a main processing thread detect the flag and process it.